### PR TITLE
Move batch error aggregation under call combiner

### PR DESCRIPTION
**Pull request by kpayson64**
This should save ~192 bytes/call

### CODE REVIEW

File: src/core/lib/surface/call.c [hunk 1]
```diff
@@ -1467,7 +1444,6 @@ static void receiving_stream_ready(grpc_exec_ctx *exec_ctx, void *bctlp,
       grpc_byte_stream_destroy(exec_ctx, call->receiving_stream);
       call->receiving_stream = NULL;
     }
-    add_batch_error(exec_ctx, bctl, GRPC_ERROR_REF(error), true);
```
Comment (kpayson64) [inline src/core/lib/surface/call.c]: I think this is safe to remove.  The only places it is called from is ```receiving_stream_ready_in_call_combiner``` and ```receiving_initial_metadata_ready```.  In both cases we already call ```add_batch_error``` with this error.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                   FILE SIZE
 ++++++++++++++ GROWING                                     ++++++++++++++

 -------------- SHRINKING                                   --------------
  -0.1%     -16 src/core/lib/surface/call.cc                    -16  -0.1%
     -13.4%    -192 finish_batch_step                              -192 -13.4%
      -4.2%     -16 receiving_stream_ready                          -16  -4.2%
      -2.5%     -11 [Unmapped]                                      -11  -2.5%

 -+-+-+-+-+-+-+ MIXED                                       +-+-+-+-+-+-+-
  +0.0%     +24 [None]                                         -128  -0.0%

  +0.0%      +8 TOTAL                                          -144  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__32k_2_0.counters.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__32k_2_0.opt.old: 5
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.opt.old: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.counters.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.counters.new: 4
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.counters.new: 4
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.opt.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.counters.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.counters.old: 4
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__32k_2_0.opt.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.opt.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.counters.old: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.opt.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.opt.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.counters.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.opt.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.counters.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.opt.old: 5
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.opt.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.opt.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.opt.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.counters.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.opt.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.counters.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.opt.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.counters.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.counters.new: 4
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.opt.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.opt.new: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.counters.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.opt.old: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__32k_2_0.opt.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.counters.new: 4
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.opt.new: 3
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.opt.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__1_2_0.counters.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__0_2_0.opt.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__64_2_0.opt.old: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__32k_2_0.counters.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__4k_2_0.opt.old: 2
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_MinInProcessCHTTP2_NoOpMutator_NoOpMutator__8_2_0.opt.new: 1
    bm_fullstack_streaming_ping_pong.BM_StreamingPingPongWithCoalescingApi_InProcessCHTTP2_NoOpMutator_NoOpMutator__512_2_0.counters.old: 6


[microbenchmarks] Performance differences noted:
Benchmark                                                                                  allocs_per_iteration    atm_add_per_iteration    atm_cas_per_iteration    cpu_time    nows_per_iteration    real_time
-----------------------------------------------------------------------------------------  ----------------------  -----------------------  -----------------------  ----------  --------------------  -----------
BM_LameChannelCallCreateCore                                                               +8%                     -9%
BM_LameChannelCallCreateCoreSeparateBatch                                                  +12%                    -7%                      +11%                                 +12%
BM_LameChannelCallCreateCpp                                                                +12%                    -7%                      +11%                                 +12%
BM_PumpStreamClientToServer<InProcess>/256k                                                                                                                          -10%                              -10%
BM_PumpStreamClientToServer<InProcess>/32k                                                                                                                           -6%                               -6%
BM_PumpStreamServerToClient<InProcess>/256k                                                                                                                          -6%                               -6%
BM_PumpStreamServerToClient<InProcess>/32k                                                                                                                           -10%                              -10%
BM_PumpStreamServerToClient<InProcessCHTTP2>/256k                                                                                                                    -5%                               -5%
BM_PumpStreamServerToClient<InProcessCHTTP2>/32k                                                                                                                     -4%                               -4%
BM_StreamingPingPong<InProcess, NoOpMutator, NoOpMutator>/256k/2                                                                                                     +6%                               +6%
BM_StreamingPingPong<MinInProcess, NoOpMutator, NoOpMutator>/256k/1                                                                                                  +5%                               +5%
BM_StreamingPingPong<MinInProcess, NoOpMutator, NoOpMutator>/256k/2                                                                                                  +6%                               +6%
BM_StreamingPingPong<MinInProcess, NoOpMutator, NoOpMutator>/32k/1                                                                                                   +5%                               +5%
BM_StreamingPingPongMsgs<InProcess, NoOpMutator, NoOpMutator>/256k                                                                                                   +6%                               +6%
BM_StreamingPingPongMsgs<InProcess, NoOpMutator, NoOpMutator>/32k                                                                                                    +11%                              +11%
BM_StreamingPingPongMsgs<InProcessCHTTP2, NoOpMutator, NoOpMutator>/32k                                                                                              +6%                               +6%
BM_StreamingPingPongMsgs<MinInProcess, NoOpMutator, NoOpMutator>/256k                                                                                                +8%                               +8%
BM_StreamingPingPongMsgs<MinInProcess, NoOpMutator, NoOpMutator>/32k                                                                                                 +9%                               +9%
BM_StreamingPingPongMsgs<MinInProcessCHTTP2, NoOpMutator, NoOpMutator>/32k                                                                                           +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/256k/1/0                                                                                  +5%                               +5%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/256k/1/1                                                                                  +5%                               +5%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/256k/2/0                                                                                  +7%                               +7%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/32k/1/1                                                                                   +5%                               +5%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/32k/2/0                                                                                   +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/32k/2/1                                                                                   +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<InProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/1/0                                                                            +4%
BM_StreamingPingPongWithCoalescingApi<InProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/2/0                                                                            +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<InProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/2/1                                                                            +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/256k/1/1                                                                               +6%                               +6%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/256k/2/0                                                                               +6%                               +6%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/256k/2/1                                                                               +6%                               +6%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/32k/1/1                                                                                +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/32k/2/0                                                                                +4%                               +4%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/32k/2/1                                                                                +5%                               +5%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/0/256k                                                                                                         +10%                              +10%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/0/2M                                                                                                           +4%                               +4%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/256k/0                                                                                                         +11%                              +11%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/256k/256k                                                                                                      +5%                               +5%
BM_UnaryPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/0/256k                                                                                                   +4%                               +4%
BM_UnaryPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/0                                                                                                   +4%                               +4%
BM_UnaryPingPong<InProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/256k                                                                                                +5%                               +5%
BM_UnaryPingPong<MinInProcess, NoOpMutator, NoOpMutator>/0/256k                                                                                                      +8%                               +8%
BM_UnaryPingPong<MinInProcess, NoOpMutator, NoOpMutator>/256k/0                                                                                                      +9%                               +9%
BM_UnaryPingPong<MinInProcess, NoOpMutator, NoOpMutator>/256k/256k                                                                                                   +6%                               +6%
BM_UnaryPingPong<MinInProcess, NoOpMutator, NoOpMutator>/32k/32k                                                                                                     +7%                               +7%
BM_UnaryPingPong<MinInProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/0                                                                                                +5%                               +5%
BM_UnaryPingPong<MinInProcessCHTTP2, NoOpMutator, NoOpMutator>/256k/256k                                                                                             +4%                               +4%
```

**Comment by ctiller** [comment]
Let's pause here: CPU increase is significant, and it might be we can get these improvements in other ways

**Comment by markdroth** [comment]
Did we ever figure out why this caused an increase in CPU usage?

**Comment by kpayson64** [comment]
No, but I don't think this change is relevant anymore.  I'll close the PR.

**Pull request closed without merging**